### PR TITLE
Update Pallas distributed tutorials with jax.make_mesh

### PR DESCRIPTION
**Pull request by justinjfu**
Replaces old mesh generation with `jax.make_mesh`.

### CODE REVIEW

File: docs/pallas/tpu/distributed.md [hunk 1]
```diff
@@ -207,8 +207,7 @@ id: YkyIKN2thZ-V
 outputId: 9b7ed142-d161-4237-fed8-cbce41adc5f0
 ---
 partition = P(None, 'x')
-devices = mesh_utils.create_device_mesh((1, num_devices))
-mesh = jax.sharding.Mesh(devices, partition)
+mesh = jax.make_mesh((1, num_devices), ('_', 'x'))
```
Comment (yashk2810) [inline docs/pallas/tpu/distributed.md:210]: This is incorrect. Make it a 1D mesh or give the axis name a proper name instead of `_`.

**Review by yashk2810 — APPROVED** [review]

File: docs/pallas/tpu/distributed.md (hunk 1 above)
Comment (justinjfu) [reply docs/pallas/tpu/distributed.md]: Done. Updated everything to 1D meshes.
